### ISSUE TITLE
JavadocVariable: add "allowedAnnotations" property

### ISSUE THREAD

**Issue by ljacqu** [checkstyle/checkstyle#4089]
Akin to the `allowedAnnotations` property on JavadocMethod, I would like to suppress the `JavadocVariable` warning for fields that have a given annotation.

For example, with a configuration:
```xml
<module name="JavadocVariable">
   <property name="allowedAnnotations" value="Property"/>
   <property name="scope" value="package"/>
</module>
```

I would expect the following behavior:
```java
@Property
public static final int TIMEOUT = 5; // no violation

/** Contains the name. */
public static String NAME = "Test"; // no violation

public static int TIMEOUT = 30; // violation
```

**Comment by sagar-shah94** [comment]
@romani Is this issue approved?what is difficulty level of this issue?

**Comment by romani** [comment]
this issue is not approved, because whole Check need to be reimplemented completely.

**Comment by Wujiaxuan007** [comment]
JavadocVariable: add "allowedAnnotations" property : (

**Comment by Prathamesh-007** [comment]
Is this issue still active?

**Comment by romani** [comment]
@gianmarcoschifone , please investigate if we can archive this be xpath filter or we do need Check extension with new property?

**Comment by gianmarcoschifone** [comment]
@romani can be done with xpath filter:

```
gianmarco@Mac ~/Documents/Projects/testing-checkstyle % cat Test.java
public class Test {
    @Property
    public static final int TIMEOUT = 5; // no violation

    /** Contains the name. */
    public static String NAME = "Test"; // no violation

    public static int TIMEOUT = 30; // violation
}

gianmarco@Mac ~/Documents/Projects/testing-checkstyle % cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="JavadocVariable">
    </module>

    <module name="SuppressionXpathSingleFilter">
      <property name="checks" value="JavadocVariable"/>
      <property name="query"
          value="//VARIABLE_DEF[./MODIFIERS/ANNOTATION/IDENT[@text='Property']]"/>
    </module>
  </module>
</module>

gianmarco@Mac ~/Documents/Projects/testing-checkstyle % java -jar checkstyle-13.9.0-SNAPSHOT-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /Users/gianmarco/Documents/Projects/testing-checkstyle/Test.java:8:5: Missing a Javadoc comment. [JavadocVariable]
Audit done.
Checkstyle ends with 1 errors.
```

**Comment by romani** [comment]
thanks a lot for investigation and workaround !

we close issue for now. until more requirements come up.
